### PR TITLE
Fix Initial Sync in Restarts

### LINKED ISSUE #2196: Perform Initial Sync If Too Far Behind With Persistent Disk
Currently, regular sync works via a recursive parent fetching mechanism. That is, upon receiving a block, if we do not have its parent, we recursively perform the same action by requesting the parent until we are fully caught up. If we are too far behind, this can hit a recursive depth limit. Instead, we can force nodes to begin initial sync if they are too far behind (this depth can be subjective, perhaps 2 or 3 epochs behind is good enough)

### PR DESCRIPTION
**Pull request by nisdas**
Sets the chainstarted field, which will allow nodes to initial sync during rolling restarts.

Resolves #2196

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Please add regression test

File: beacon-chain/sync/querier.go [hunk 1]
```diff
@@ -186,10 +187,10 @@ func (q *Querier) RequestLatestHead() {
 // IsSynced checks if the node is currently synced with the
 // rest of the network.
 func (q *Querier) IsSynced() (bool, error) {
-	if !q.chainStarted {
+	if q.atGenesis {
 		return true, nil
 	}
-	if q.atGenesis {
+	if !q.chainStarted {
```
Comment (prestonvanloon) [inline beacon-chain/sync/querier.go]: can you revert the swapping of these conditions?

**Comment by nisdas** [comment]
@prestonvanloon made the changes, PTAL

File: beacon-chain/sync/service.go [hunk 2]
```diff
@@ -4,11 +4,13 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+
+	"github.com/prysmaticlabs/prysm/shared/params"
+
```
Comment (prestonvanloon) [inline beacon-chain/sync/service.go]: Space

**Pull request merged**